### PR TITLE
Set trainer.save_model state_dict format to safetensors

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change
bug ticket: https://jira.devtools.intel.com/browse/NLPTOOLKIU-1189
root cause, transformers 4.37.2  limits pytorch_model.bin loading by https://github.com/huggingface/transformers/blob/v4.37.2/src/transformers/modeling_utils.py#L530

Due to trainer.quantize with pytorch fx mode, the state_dict not only includes weight key and value but also has inc quantization config, so still keep the "pytorch_model.bin".
The state dict will update to "model.safetensors" for others situation. 
remove transformers version lower than 4.22 support.

## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by xin3he — APPROVED** [review]

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1420/

**Review by PenghuiCheng — APPROVED** [review]

**Comment by VincyZhang** [comment]
Still get the error when running accuracy:
image

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1421/

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1421/

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1426/

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/4552/ extension pass
